compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -240,9 +240,21 @@ OMR::X86::CodeGenerator::initialize(TR::Compilation *comp)
 
    if (TR::Compiler->target.cpu.getX86SupportsTM(comp) && !comp->getOption(TR_DisableTM))
       {
-      if (TR::Compiler->target.is64Bit())
-         self()->setSupportsTM(); // disable tm on 32bits for now
-      }
+
+	 /**
+	  * Due to many verions of Haswell and a small number of Broadwell have defects for TM and then disabled by Intel, 
+	  * we will return false for any versions before Broadwell.
+	  *
+	  * TODO: Need to figure out from which mode of Broadwell start supporting TM
+	  */
+      if (!_targetProcessorInfo.isIntelHaswell())
+         {
+         if (TR::Compiler->target.is64Bit())
+            {
+            self()->setSupportsTM(); // disable tm on 32bits for now
```

Comment:
my remark was only about the fact that the comment itself is not informative. The code itself tells me that it is disabled for 32-bit; i don't need a comment to tell me that. What is warranted, IMO, is an explanation for why it is disabled on 32-bits. I wasn't suggesting we change the code at all.

If we have no additional information on why it's disabled for 32-bits, then I guess we can just leave it as-is.